### PR TITLE
[EMCAL-704] Add debug printouts for subevents and filter deadbeaf messages

### PR DESCRIPTION
**Pull request by mfasDa**
For the initial debugging phase adding printouts
for the digits received from subevents.

### CODE REVIEW

**Comment by mfasDa** [comment]
Adding filter for deadbeaf messages introduced in https://github.com/AliceO2Group/AliceO2/pull/6772. @shahor02 Is the filter implemented correctly?

File: Modules/EMCAL/include/EMCAL/DigitsQcTask.h [hunk 1]
```diff
@@ -110,8 +112,13 @@ class DigitsQcTask final : public TaskInterface
         nObjects += ev.mCellRange.getEntries();
```
Comment (shahor02) [inline Modules/EMCAL/include/EMCAL/DigitsQcTask.h:112]: add {} for if.

File: Modules/EMCAL/src/DigitsQcTask.cxx [hunk 2]
```diff
@@ -108,15 +108,21 @@ void DigitsQcTask::monitorData(o2::framework::ProcessingContext& ctx)
...
-  std::unordered_map<int, gsl::span<const o2::emcal::TriggerRecord>> triggerRecordSubevents;
+  std::unordered_map<header::DataHeader::SubSpecificationType, gsl::span<const o2::emcal::Cell>> cellSubEvents;
+  std::unordered_map<header::DataHeader::SubSpecificationType, gsl::span<const o2::emcal::TriggerRecord>> triggerRecordSubevents;
 
   for (const auto& celldata : framework::InputRecordWalker(ctx.inputs(), cellInputs)) {
-    int subspecification = framework::DataRefUtils::getHeader<header::DataHeader*>(celldata)->subSpecification;
+    auto subspecification = framework::DataRefUtils::getHeader<header::DataHeader*>(celldata)->subSpecification;
+    // Discard message if it is a deadbeaf message (empty timeframe)
+    if (subspecification == 0xdeadbeaf)
+      continue;
```
Comment (shahor02) [inline Modules/EMCAL/src/DigitsQcTask.cxx:118]: if {..}

File: Modules/EMCAL/src/DigitsQcTask.cxx [hunk 3]
```diff
@@ -108,15 +108,21 @@ void DigitsQcTask::monitorData(o2::framework::ProcessingContext& ctx)
...
     cellSubEvents[subspecification] = ctx.inputs().get<gsl::span<o2::emcal::Cell>>(celldata);
   }
   for (const auto& trgrecorddata : framework::InputRecordWalker(ctx.inputs(), triggerRecordInputs)) {
-    int subspecification = framework::DataRefUtils::getHeader<header::DataHeader*>(trgrecorddata)->subSpecification;
+    auto subspecification = framework::DataRefUtils::getHeader<header::DataHeader*>(trgrecorddata)->subSpecification;
+    // Discard message if it is a deadbeaf message (empty timeframe)
+    if (subspecification == 0xdeadbeaf)
+      continue;
```
Comment (shahor02) [inline Modules/EMCAL/src/DigitsQcTask.cxx:125]: if {}

**Review by shahor02 — COMMENTED** [review]
@mfasDa Looks ok, except the `if {}` comments. Since your `DigitsQcTask::monitorData` does not send out any message, you just have to filter out eventual 0xdeadbeef messages.

**Comment by mfasDa** [comment]
OK, brackets are fixed. @knopers8 could you please merge once the tests pass?

**Review by knopers8 — APPROVED** [review]

**Pull request merged**